pkg/syncer/storagepool/service.go

```diff
@@ -94,11 +101,35 @@ func InitStoragePoolService(ctx context.Context, configInfo *commontypes.ConfigI
 		return err
 	}
 
+	migrationController := initMigrationController(vc, configInfo.Cfg.Global.ClusterID)
+	go func() {
+		diskDecommEnablementTicker := time.NewTicker(common.DefaultFeatureEnablementCheckInterval)
+		defer diskDecommEnablementTicker.Stop()
+		clusterFlavor := cnstypes.CnsClusterFlavorWorkload
+		for ; true; <-diskDecommEnablementTicker.C {
+			k8sInitParams := k8sorchestrator.K8sSupervisorInitParams{
+				SupervisorFeatureStatesConfigInfo: configInfo.Cfg.FeatureStatesConfig,
+			}
+			coCommonInterface, _ := commonco.GetContainerOrchestratorInterface(ctx, common.Kubernetes, clusterFlavor, k8sInitParams)
+			if !coCommonInterface.IsFSSEnabled(ctx, common.VSANDirectDiskDecommission) {
+				log.Infof("VSANDirectDiskDecommission feature is disabled on the cluster")
+			} else {
+				_, err := initDiskDecommController(ctx, migrationController)
+				if err != nil {
+					log.Warnf("Error while initializing disk decommission controller. Error: %+v. Retry will be triggered at %v", err, time.Now().Add(common.DefaultFeatureEnablementCheckInterval))
+					continue
+				}
+				break
+			}
+		}
```

Comment:
@divyenpatel @SandeepPissay It seems that csi wcp e2e tests pipeline is broken. I checked health of the pipeline by running wcp tests on an empty PR. End result of the pipeline was `Failed` https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/452#issuecomment-713472678. 
The changes introduced in the PR is behind a feature state switch, which is currently off. So this PR will not introduce any regression. I kindly ask you to approve these changes.